### PR TITLE
do not differentiate through the construction of BitArray

### PR DESCRIPTION
**Pull request by frankschae**
Based on @oxinabox 's reply here: https://github.com/SciML/DiffEqFlux.jl/issues/502#issuecomment-902023817 .

This solves also the issues with the callback construction in https://github.com/SciML/DiffEqFlux.jl/issues/502.

Does it need a test? Should I maybe add one from https://github.com/SciML/DiffEqFlux.jl/issues/502#issuecomment-902015247 in here: https://github.com/JuliaDiff/ChainRules.jl/blob/master/test/rulesets/Base/array.jl ?

### CODE REVIEW

**Comment by oxinabox** [comment]
Cool, LGTM.
We don't generally test `@non_differentiable` rules,
because any bugs in them are not normally in the rule, but in the definition of the `@non_differentiable` macro.
And that has it's own tests.

Sometimes we test them, if we have had them defined wrong in the past.
But that is not the case here

**Comment by oxinabox** [comment]
Will merge once CI passes.
(and I will bump the version and tag a release).
Or you can bump the version and make it easier for me to tag a release.
This is a new feature so it is a minor bump.

**Comment by frankschae** [comment]
cool, Thanks a lot!

File: Project.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name = "ChainRules"
 uuid = "082447d4-558c-5d27-93f4-14fc19e9eca2"
-version = "1.9"
+version = "1.9.1"
```
Comment (oxinabox) [inline Project.toml:3]: Minor change.
Those is a feature not a bug fix


```suggestion
version = "1.10.0"
```

File: src/rulesets/Base/nondiff.jl [hunk 2]
```diff
@@ -84,6 +84,7 @@
 @non_differentiable Vector(::AbstractArray{Bool})
 
 @non_differentiable Array(::AbstractArray{Bool})
+@non_differentiable BitArray(::Any)
```
Comment (mzgubic) [inline src/rulesets/Base/nondiff.jl:87]: Is there a reason we didn't splat this?
```julia
@non_differentiable BitArray(::Any...)
```

Comment (oxinabox) [reply src/rulesets/Base/nondiff.jl:87]: Doesn't it only have 1 constructor?
I didn't actually check

Comment (mzgubic) [reply src/rulesets/Base/nondiff.jl:87]:
```julia
julia> methods(BitArray)
# 6 methods for type constructor:
[1] BitArray(::UndefInitializer, dims::Integer...) in Base at bitarray.jl:69
[2] BitArray(::UndefInitializer, dims::Tuple{Vararg{Integer, N}}) where N in Base at bitarray.jl:71
[3] BitArray(x::BitArray) in Base at bitarray.jl:543
[4] BitArray(A::AbstractArray{var"#s79", N} where var"#s79") where N in Base at bitarray.jl:503
[5] (::Type{T})(x::T) where T<:BitArray in Base at bitarray.jl:542
[6] BitArray(itr) in Base at bitarray.jl:574
```
there seems to be more

Comment (frankschae) [reply src/rulesets/Base/nondiff.jl:87]: Should I make another PR?

Comment (mzgubic) [reply src/rulesets/Base/nondiff.jl:87]: Please do, that would be great, thanks!

**Pull request merged**